### PR TITLE
[perl/en] more perlish iterations on perl5

### PR DESCRIPTION
**Pull request by centreti**
- [x] PR touches only one file (or a set of logically related files with similar changes made)
- [ ] Content changes are aimed at _intermediate to experienced programmers_ (this is a poor format for explaining fundamental programming concepts)
- [x] YAML Frontmatter formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Seriously, look at it now. Watch for quotes and double-check field names.

C-style for-loops aren't Perlish, and should not be encouraged.

Also I added a `map` version, because it's worth knowing it seeing as how it's the most efficient operator for lists/arrays.

### CODE REVIEW

File: perl.html.markdown [hunk 1]
```diff
@@ -126,18 +126,20 @@ while (condition) {
 
 
 # for loops and iteration
-for (my $i = 0; $i < $max; $i++) {
+for my $i (0 .. $max) {
   print "index is $i";
 }
 
-for (my $i = 0; $i < @elements; $i++) {
+while (my $i = each @elements) {
```
Comment (vendethiel) [inline perl.html.markdown]: I don't remember seeing that for an array. Is it really idiomatic? Also might warrant a version using `$#`

Comment (centreti) [inline perl.html.markdown]: It isn't really idiomatic, but rather accomplishes the same thing. Honestly, I wouldn't really know of any reason to iterate over an array in this way to begin with. Might make sense just to drop this entirely.

Comment (vendethiel) [inline perl.html.markdown]: Sounds about right...

Comment (Grinnz) [inline perl.html.markdown]: A more idiomatic way to do this is:
`for (0..$#elements) {`

**Comment by vendethiel** [comment]
Thanks!

**Pull request merged**